.gitbook/assets/ReadonlyREST-api.yml

```diff
@@ -0,0 +1,209 @@
+                ImpersonationNotAllowedResponseExample:
+                  $ref: "#/components/examples/ImpersonationNotAllowedResponseExample"
+                RorIsNotStartedYetResponseExample:
+                  $ref: "#/components/examples/RorIsNotStartedYetResponseExample"
+                RorFailedToStartResponseExample:
+                  $ref: "#/components/examples/RorFailedToStartResponseExample"
+                CannotSaveClusterWideSettingsOnFreeLicense:
+                  $ref: "#/components/examples/CannotSaveClusterWideSettingsOnFreeLicense"
+
+components:
+  securitySchemes:
+    http:
+      type: http
+      scheme: basic
+
+  schemas:
+    ForbiddenRorResponse:
+      type: object
+      required:
+        - error
+        - status
+      properties:
+        error:
+          type: object
+          required:
+            - type
+            - reason
+            - due_to
+          properties:
+            type:
+              type: string
+              enum:
+                - forbidden_response
+            reason:
+              type: string
+              enum:
+                - forbidden
+            due_to:
+              type: array
+              description: |
+                It's an array of causes why ReadonlyREST's ACL forbade the request. 
+                WARNING: Please notice, that one element array ES used to print as a string!!!
+              items:
+                type: string
+                enum:
+                  - OPERATION_NOT_ALLOWED
+                  - FORBIDDEN_BY_BLOCK
+                  - TEST_SETTINGS_NOT_CONFIGURED
+                  - IMPERSONATION_NOT_SUPPORTED
+                  - IMPERSONATION_NOT_ALLOWED
+                  - READONLYREST_NOT_READY_YET
+                  - READONLYREST_NOT_ENABLED
+                  - READONLYREST_FAILED_TO_START
+                  - cannotSaveClusterWideSettingsOnFreeLicense
+        status:
+          type: number
+          example: 403
+
+    XRorCorrelationId:
+      type: string
+      description: "Correlation ID to correlate requests/responses logs within one ROR KBN session"
+      example: "CED054A0-9881-401E-AEED-4FB4C5915DD6"
+
+    UpdateMainSettingsRequest:
+      type: object
+      required:
+        - settings
+      properties:
+        settings:
+          type: string
+          description: "ReadonlyREST main settings YAML"
+          example: |
+            readonlyrest:\\r\\n    access_control_rules:\\r\\n\\r\\n    - name: | \\""Require HTTP Basic | Auth\\""\\r\\n      type: allow\\r\\n      indices: | [\\""test\\""]\\r\\n\\r\\n    - name: | \\""other\\""\\r\\n      auth_key: \\""admin:test\\""
+
+    UpdateMainSettingsResponse:
+      type: object
+      required:
+        - status
+        - message
+      properties:
+        status:
+          type: string
+          description: Update Main Settings status
+          enum:
+            - ok
```

Comment:
Let's change them to something more "normal". Eg. "SUCCESS", "FAILURE" (instead of "ok", "ko")